module/polls/helper.js

```diff
@@ -0,0 +1,631 @@
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_QUESTIONS_FETCHED,
+                    data: result
+                });
+
+                }
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+     /**
+    * Poll report.
+    * @method
+    * @name report
+    * @param {String} pollId - pollId 
+    * @returns {Object} - Poll report data
+    */
+
+   static report(pollId = "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if (pollId == "") {
+                throw new Error (messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK)
+            }
+
+            let pollDocument = await this.pollDocuments
+            (
+                { _id : pollId },
+                [
+                    "result",
+                    "questions",
+                    "numberOfResponses"
+                ]
+            )
+
+            if (!pollDocument.length) {
+                throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+            }
+
+            if (pollDocument[0].numberOfResponses == 0) {
+                throw new Error(messageConstants.apiResponses.POLL_SUBMISSIONS_NOT_FOUND)
+            }
+            
+            let reports = [];
+
+            let report = {
+                chart: {
+                    type: 'bar'
+                },
+                title: {
+                    text: ''
+                },
+                accessibility: {
+                    announceNewData: {
+                        enabled: false
+                    }
+                },
+                xAxis: {
+                    type: 'category'
+                },
+                legend: {
+                    enabled: false
+                },
+                credits: {
+                    enabled: false
+                },
+                plotOptions: {
+                    series: {
+                        borderWidth: 0,
+                        dataLabels: {
+                            enabled: true,
+                            format: '{point.y:.1f}%'
+                        }
+                    }
+                },
+                series: [
+                    {
+                        colorByPoint: true,
+                        data: []
+                    }
+                ]
+            }
+            
+            let groupByQuestionId = _.keyBy(pollDocument[0].questions, 'qid');
+            let result = pollDocument[0].result;
+
+            Object.keys(result).forEach( singleQuestion => {
+                
+                let groupOptionsByValue = _.keyBy(groupByQuestionId[singleQuestion].options,"value");
+
+                report.title.text = groupByQuestionId[singleQuestion].question;
```

Comment:
@deepakr123 - How will the report title be set if more than 1 question is present ?

Reply:
resolved